### PR TITLE
Replace InstanceKey.StringCode() and DisplayString() with String()

### PR DESCRIPTION
**Pull request by sjmudd**
The 3 functions generated the same output and were interchangeable so the code is cleaner and easier to understand if just String() is used.  This suggested patch removes the redundant functions.

It's confusing otherwise and you need to check the code to see what the difference is between the 3 functions when actually there's no difference at all.  There are no functional changes to any of the code.

- [x] contributed code is using same conventions as original code
- [x] code is formatted via `gofmt` (please avoid `goimports`)
- [x] code is built via `./build.sh`

### CODE REVIEW

**Comment by sjmudd** [comment]
Shlomi, I saw the travis build failed and fixed the failed tests, and have pushed these fixes but don't see a way to do an "update branch".   The workflow here is not clear to me.

**Comment by shlomi-noach** [comment]
@sjmudd I updated the branch for you. It's strange that you don't see the button, since you're a committer.

**Comment by shlomi-noach** [comment]
@sjmudd the reason for the different functions:

- `String()` is by convention a visual string representation of an object; something that is made for humans and debug output.
- As such, I do not wish to rely on `String()` on anything that is logic. So never use it as a key to a map.
- Which is what `StringCode()` is there for.
- Granted, both currently return the same value.
- I'm happy to remove `DisplayString()` though.

In summary, I will not replace `StringCode()` with `String()` but am happy to replace `DisplayString()` with `String()`.

**Comment by sjmudd** [comment]
@shlomi-noach Thanks for clarifying. Perhaps this could be clearer in the code comments?

It's easier to close this PR and create a new one just changing DisplayString() to String() and I'll add a comment along the lines of what you indicate so this is clear.
